provider/provider.go

```diff
@@ -1338,7 +1344,9 @@ func (s *SweepingProvider) provideRegions(regions []keyspace.Region, addrInfo pe
 			}
 			continue
 		}
-		s.provideCounter.Add(s.ctx, int64(len(allKeys)))
+		keyCount := len(allKeys)
+		s.provideCounter.Add(s.ctx, int64(keyCount))
+		logger.Debugw("sent provider records", "prefix", r.Prefix, "count", keyCount, "keys", allKeys)
```

Comment:
💭  this can get pretty hardcore in debug level, 10748 keys in one line:

```
2025-09-27T03:35:58.329Z	DEBUG	dht/provider	provider/provider.go:1336	sent provider records	{"prefix": "01001101", "count": 10748, "keys": ["EiB9gQIoN[...]
```